code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstancePreStart.java

```diff
@@ -58,6 +62,12 @@ public void doWithLockNoResult() {
                     });
                 }
             }
+            Map<String, Object> driver_opts = DataAccessor.fieldMap(v, VolumeConstants.FIELD_VOLUME_DRIVER_OPTS);
+            if (driver_opts.containsKey(VolumeConstants.EC2_AZ)) {
+                String zone = driver_opts.get(VolumeConstants.EC2_AZ).toString();
+                labels.put(VolumeConstants.SCALE_GROUP_LABEL, String.format("key=%s;%s=1", VolumeConstants.EC2_AZ_HOST_LABEL_KEY, zone));
```

Comment:
Why do we have to do scale_per_group scheduling here? Shouldn't a simple hard affinity constraint to hosts with the matching AZ label work?